ui/packages/consul-ui/app/components/consul/bucket/list/index.js

```diff
@@ -71,15 +71,13 @@ export default class ConsulBucketList extends Component {
       return [nspaceItem];
     }
 
+    if (peerPart.length && abilities.can('use nspaces')) {
+      return [nspaceItem];
+    }
+
     if (nspace && abilities.can('use nspaces')) {
       if (item.Namespace !== nspace) {
-        return [
-          {
-            type: 'nspace',
-            label: 'Namespace',
-            item: item.Namespace,
-          },
-        ];
+        return [nspaceItem];
```

Comment:
Would it be correct to assume that the structure of `nspaceItem` the same as the object that was being returned previously (or that it at least has the same keys)?

Reply:
yes - we could have returned `nspaceItem` before. I realized that when looking at the code again today.